assets/js/base/utils/lazy-load-script.ts

```diff
@@ -18,6 +31,23 @@ const isScriptTagInDOM = ( scriptId: string ): boolean => {
 	return scriptElements.length > 0;
 };
 
+/**
+ * Appends a script element to the document body if an script with the same id
+ * doesn't exist.
+ */
+const appendScript = ( attributes: appendScriptAttributesParam ): void => {
+	// Abort if id is not valid or a script with the same id exists.
+	if ( ! isString( attributes.id ) || isScriptTagInDOM( attributes.id ) ) {
+		return;
+	}
+	const scriptElement = document.createElement( 'script' );
+	for ( const attr in attributes ) {
+		const key = attr as keyof appendScriptAttributesParam;
+		scriptElement[ key ] = attributes[ key ];
+	}
+	document.body.appendChild( scriptElement );
+};
+
```

Comment:
:point_up_2: This is the code that matters. Given an object of attributes, it creates an `<script>` element with those attributes. It works great, but TypeScript shows an error in line 46:
```JS
scriptElement[ key ] = attributes[ key ];
```
The error:
> Type 'string | OnErrorEventHandlerNonNull | (() => void) | undefined' is not assignable to type 'string & OnErrorEventHandlerNonNull & ((this: GlobalEventHandlers, ev: Event) => any)'.
>  Type 'undefined' is not assignable to type 'string & OnErrorEventHandlerNonNull & ((this: GlobalEventHandlers, ev: Event) => any)'.
>    Type 'undefined' is not assignable to type 'string'.ts(2322)

I feel that TypeScript gets confused by the `for ... in` and it warns that we might be using the wrong attribute values in a script tag.

Some things that I tried:
* Replacing `for ... in` with `Object.keys()` or `Object.entries()`. The issue persists.
* Explicitly setting the `attributes[ key ]` type, but I don't know how to do that inside a loop. We would need to read the type of each specific attribute from `appendScriptAttributesParam`.
* Using [`ScriptHTMLAttributes`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1349b640d4d07f40aa7c1c6931f18e3fbf667f3a/types/react/index.d.ts#L2228) from DefinitelyTyped. I couldn't get it to work, and those types are React-oriented (`onerror` would be `onError`, `innerHTML` doesn't exist...) so probably not something we can use in this case.

Any other ideas on how to solve this? :slightly_smiling_face: